org.eclipse.xtext.ui.tests/src-longrunning/org/eclipse/xtext/ui/tests/editor/DirtyStateEditorSupportIntegrationTest.java

```diff
@@ -116,7 +124,72 @@ public void testUndoRedo() throws Exception {
 				.getQualifiedName().getLastSegment());
 		Iterables.getLast(events);
 	}
+	
+	/*
+	 * @see https://github.com/eclipse/xtext/issues/2385
+	 */
+	@Test public void testModifyFileInExternEditor() throws Exception {
+		IXtextDocument document = editor.getDocument();
+
+		Display.getDefault().readAndDispatch();
+		
+		assertNotEquals(document.get(), "");
+		assertTrue(editor.getDocumentProvider() instanceof FileDocumentProvider);
+		
+		FileDocumentProvider fileDocumentProvider = (FileDocumentProvider)editor.getDocumentProvider();
+		
+		getWorkbench().getActiveWorkbenchWindow().getActivePage().getViewReferences()[0].getView(false).setFocus();
+		
+		File externalEditFile = ((IFileEditorInput)editor.getEditorInput()).getFile().getLocation().toFile();
+		try (FileWriter fw = new FileWriter(externalEditFile)) {
+			fw.write("");
+		}
+		
+		assertFalse(fileDocumentProvider.isSynchronized(editor.getEditorInput()));
+
+		editor.setFocus();
+		Job.getJobManager().join(ResourcesPlugin.FAMILY_AUTO_REFRESH, null);
+		syncUtil.yieldToQueuedDisplayJobs(new NullProgressMonitor());
+		syncUtil.waitForReconciler(editor);
+		syncUtil.waitForBuild(new NullProgressMonitor());
+		
+		assertTrue(fileDocumentProvider.isSynchronized(editor.getEditorInput()));
+		
+		assertEquals(document.get(), "");
+	}
 
+	/**
+	 * This test needs a manual button click.
+	 */
+	@Ignore
```

Comment:
When I get this dialog:

<img width="685" alt="Screenshot 2024-09-04 at 08 52 51" src="https://github.com/user-attachments/assets/32ceb4e5-6402-4ec1-a3c6-a0f3613ab312">

I'd expect that `Replace editor content` sets the editor content to the text in the file. But apparently it'll override the file change on disk. That's not ok.

Reply:
really? I can't test It yet, as far as i remember it discards the changes in the editor.. Its not supposed to write anything on the disc..

Reply:
Btw. You have to ignore the changes and then the changes in XtextEditor is necessary..

Reply:
> really? I can't test It yet, as far as i remember it discards the changes in the editor.. Its not supposed to write anything on the disc..

Yes, I tried this a couple of times and added additional assertions to confirm my observation.

Reply:
Might be that this is due to the doSave() though. Testing again.

Reply:
Ok, test works as expected, indeed.

Reply:
Thanks for clarification.. Is this PR ok now, or is there something else what should be checked?

Reply:
@szarnekow is it possible to merge this PR today? :)

Reply:
Yes, it is. @iloveeclipse can you confirm that it works for you guys?

Reply:
It works for me..

Reply:
> Yes, it is. @iloveeclipse can you confirm that it works for you guys?

Mehmet is in my team, and thanks for asking.

Reply:
Excellent. So let's move this forward. Thank you for the fix and the patience with the review process @mehmet-karaman

Reply:
@szarnekow Thank you for your review and yours hints..